README.md

```diff
@@ -397,6 +397,8 @@ where:
      if full primary key equal condition is specified
   * `timeout` (`?number`) - `vshard.call` timeout (in seconds)
   * `fields` (`?table`) - field names for getting only a subset of fields
+  * `fullscan` (`?boolean`) - if `true` then a critical log entry will be skipped
+    on potentially long `select` (`>`, `>=`, `<`, `<=` and `first` > 1000)
```

Comment:
```offset=N``` just iterates over an index N times. ```after``` acts in other way: iterates over an index with GE or LE from the tuple. So we don't know "offset" of ```after```.

But yes, firs t!= limit, it can fetch a lot more tuples. But it seems to me that if we start to care about it, it will be too difficult to understand for an user. first <= 1000 is very simple and obvious condition.

Reply:
Yeah, I understand that it's really difficult to implement. Just decided to clarify this case explicitly.

Reply:
> `after` acts in other way: iterates over an index with GE or LE from the tuple.

If it is so in all cases (I don't know), we should not count it. We should count only tuples, which are linearly traversed (it is why the `offset` value is counted).

BTW, it would be good to leave some internal documentation regarding the query planner fields. Their meaning is not obvious.

Say, what the executor will do, when both 'scan value' and 'after scan value' are present? I guess, it'll use 'after scan value' with GT/LT and transform 'scan value' to a post-filter, but I'm not completely sure.

Whether the 'after scan value' always corresponds to the index determined by the first select condition? What if the first condition is not backed by any index? Will the 'after' condition be transformed to a post-filter (rather than GT/LT on an index) in the case?

Now it is hard to think about different cases, to be honest.

Reply:
I think it worth to write a developer documentation on the topic. I filed https://github.com/tarantool/crud/issues/292.

Reply:
I described the algorithm in #292.

In short, we can consider `after` as a regular full scan. Sometimes we can additionally skip some tuples in `after` case (if a key from condition == a key from after), but I think it's not critical. In any case, there is no way to calculate amount of the tuples in advance.

Reply:
It looks as something that we'll completely resolve in PR #295. Closing the discussion so.